Minor tweaks of the Local add-on testing page (#1175)
home-assistant/developers.home-assistant: docs/add-ons/testing.md

The fastest and recommended way to develop add-ons is using a local Visual Studio Code devcontainer. We maintain a [devcontainer for this purpose](https://github.com/home-assistant/devcontainer) which is used in all our add-ons repositories. This devcontainer setup for VS Code which will run Supervisor and Home Assistant, with all of the add-ons mapped as Local Add-ons inside, making it simple for add-on developers on Windows, Mac and Linux desktop OS-es.

- Follow the instructions to download and install the [Remote Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) VS Code extension.

> - Copy the [`devcontainer.json` file](https://github.com/home-assistant/devcontainer/raw/main/addons/devcontainer.json) to `.devcontainer/devcontainer.json` in your repository.

Reviewer: ```suggestion
- Copy the [`devcontainer.json` file](https://github.com/home-assistant/devcontainer/raw/main/addons/devcontainer.json) to `.devcontainer.json` in your repository.
```

Reviewer: This no longer needs to be in a `.devcontainer` directory

Author: I'd like to keep this consistent with at least the sample add-on repo. If this is the way I'd open a PR there.

Reviewer: please do

> - Copy the [`devcontainer.json` file](https://github.com/home-assistant/devcontainer/raw/main/addons/devcontainer.json) into either the root or the `.devcontainer/` folder of your repository.

Reviewer: ```suggestion
- Copy the [`devcontainer.json` file](https://github.com/home-assistant/devcontainer/raw/main/addons/devcontainer.json) to `.devcontainer.json` in your repository.
```
Adding `devcontainer.json` to the root does not work, it has to be `.devcontainer.json` or `.devcontainer/devcontainer.json`

If you don't want to use the devcontainer environment, you can still run add-ons locally with Docker.

> Create a new folder for data and add a test `options.json` file. After that you can run your add-on with:

Reviewer 2: Shouldn't this be the `options.yaml` file to be consistent with the rest of the documentation nowadays?

Reviewer: No, this is the file that is written by the supervisor inside the add-on
We should replace this with `bashio`

Reviewer 2: Good morning @frenck... 🤦

Author: > We should replace this with `bashio`

Isn't the `options.json` file required in every case anyway? From my understanding, `bashio` could parse the file, but it still needs to be present, doesn't it?

Reviewer 2: `bashio` doesn't use the file anymore :)
It queries the Supervisor API instead.

Author: 🤯🔥
